extensions/samples/electron-webpack/src/main.ts

```diff
@@ -0,0 +1,92 @@
+import { app, BrowserWindow } from "electron";
+import * as path from "path";
+import {
+  ElectronAuthenticator,
+  MsalElectronConfig,
+ } from '@microsoft/mgt-electron-provider/dist/Authenticator';
+ import { 
+  DataProtectionScope,
+  PersistenceCreator,
+  PersistenceCachePlugin,
+  IPersistence
+} from "@azure/msal-node-extensions";
+
+// You can use the helper functions provided through the Environment class to construct your cache path
+// The helper functions provide consistent implementations across Windows, Mac and Linux.
+const cachePath = "./cache.json";
+
+const persistenceConfiguration = {
+  cachePath,
+  dataProtectionScope: DataProtectionScope.CurrentUser,
+  serviceName: "test-msal-electron-service",
+  accountName: "test-msal-electron-account",
+  usePlaintextFileOnLinux: false,
+}
+
+function createWindow() {
+  // Create the browser window.
+  const mainWindow = new BrowserWindow({
+    height: 600,
+    webPreferences: {
+      nodeIntegration: true,
+      contextIsolation: false,
+    },
+    width: 800,
+  });
+
+  PersistenceCreator
+  .createPersistence(persistenceConfiguration)
+  .then(async (persistence: IPersistence) => {
+    // Initialize the electron authenticator
+    const config: MsalElectronConfig = {
+      clientId: '<CLIENT_ID>',
+      mainWindow: mainWindow, //BrowserWindow instance that requires auth
+      scopes: [
+        'user.read',
+        'people.read',
+        'user.readbasic.all',
+        'contacts.read',
+        'presence.read.all',
+        'presence.read',
+        'user.read.all',
+        'calendars.read',
+        'Sites.Read.All',
+        'Sites.ReadWrite.All',
```

Comment:
Just to confirm, are all these scopes actually necessary? I couldn't see what endpoints are being called

Reply:
Yes, since we are using Graph toolkit all the scopes are required by the components we are using from Graph toolkit.